src/temp_sensor.rs

```diff
@@ -0,0 +1,149 @@
+    Default,
+    XTAL,
+    RcFast,
+}
+
+#[cfg(not(any(esp32)))]
+impl From<TemperatureSensorClockSource> for temperature_sensor_clk_src_t {
+    fn from(value: TemperatureSensorClockSource) -> Self {
+        match value {
+            TemperatureSensorClockSource::XTAL => {
+                soc_periph_temperature_sensor_clk_src_t_TEMPERATURE_SENSOR_CLK_SRC_XTAL
+            }
+            TemperatureSensorClockSource::RcFast => {
+                soc_periph_temperature_sensor_clk_src_t_TEMPERATURE_SENSOR_CLK_SRC_RC_FAST
+            }
+            TemperatureSensorClockSource::Default => {
+                soc_periph_temperature_sensor_clk_src_t_TEMPERATURE_SENSOR_CLK_SRC_DEFAULT
+            }
+        }
+    }
+}
+
+#[cfg(not(any(esp32)))]
+impl From<temperature_sensor_clk_src_t> for TemperatureSensorClockSource {
+    fn from(value: temperature_sensor_clk_src_t) -> Self {
+        match value {
+            #[allow(non_upper_case_globals)]
+            soc_periph_temperature_sensor_clk_src_t_TEMPERATURE_SENSOR_CLK_SRC_XTAL => Self::XTAL,
+            #[allow(non_upper_case_globals)]
+            soc_periph_temperature_sensor_clk_src_t_TEMPERATURE_SENSOR_CLK_SRC_RC_FAST => {
+                Self::RcFast
+            }
+            // TODO: Perhaps the default value should be mapped explicitly
+            // and all other (u32) values should cause a failure
+            _ => Self::Default,
+        }
+    }
+}
+
+// -- TemperatureSensorConfig --
+
+#[cfg(not(any(esp32)))]
+#[derive(Copy, Clone)]
+/// Rust wrapper for `temperature_sensor_config_t`
+pub struct TemperatureSensorConfig {
+    // TODO: check int size
+    pub range_min: i32,
+    pub range_max: i32,
+    pub clk_src: TemperatureSensorClockSource,
+}
+
+#[cfg(not(any(esp32)))]
+impl From<temperature_sensor_config_t> for TemperatureSensorConfig {
+    fn from(value: temperature_sensor_config_t) -> Self {
+        TemperatureSensorConfig {
+            range_min: value.range_min,
+            range_max: value.range_max,
+            clk_src: value.clk_src.into(),
+        }
+    }
+}
+
+#[cfg(not(any(esp32)))]
+impl Into<temperature_sensor_config_t> for TemperatureSensorConfig {
+    fn into(self) -> temperature_sensor_config_t {
+        temperature_sensor_config_t {
+            clk_src: self.clk_src.into(),
+            range_max: self.range_max,
+            range_min: self.range_min,
+        }
+    }
+}
+
+#[cfg(not(any(esp32)))]
+impl Default for TemperatureSensorConfig {
+    fn default() -> Self {
+        TemperatureSensorConfig {
+            range_min: -10,
+            range_max: 80,
+            clk_src: TemperatureSensorClockSource::Default,
+        }
+    }
+}
+
+// -- TemperatureSensorDriver --
+
+#[cfg(not(any(esp32)))]
+pub struct TemperatureSensorDriver {
+    ptr: temperature_sensor_handle_t,
+}
+
+#[cfg(not(any(esp32)))]
+impl TemperatureSensorDriver {
+    pub fn new(config: TemperatureSensorConfig) -> Result<Self, EspError> {
```

Comment:
Take the config by `&` as we do everywhere else. You might need to change your `From` trait impl after that.

Reply:
Done(ish) in https://github.com/esp-rs/esp-idf-hal/pull/337/commits/4d34e0e17ba7a2c9c089ef614d5e55539a188e22

See https://github.com/esp-rs/esp-idf-hal/pull/337#discussion_r1749987951 for further discussion.